### PR TITLE
KM215: Validate cluster declaration's clusterRootDomain

### PR DESCRIPTION
**Pull request by yngvark**
Adds stricter validation of clusterRootDomain.

## Description
See https://trello.com/c/OVTiSgFr/215-okctl-apply-cluster-validate-clusteryaml-properly

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the release notes (for the next release).

### CODE REVIEW

**Review by paulbes — APPROVED** [review]
NewCluster() love it ❤️

File: pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go [hunk 1]
```diff
@@ -62,7 +64,10 @@ type Cluster struct {
 // Validate calls each members Validate function
 func (c Cluster) Validate() error {
 	return validation.ValidateStruct(&c,
-		validation.Field(&c.ClusterRootDomain, validation.Required, is.LowerCase),
+		validation.Field(&c.ClusterRootDomain,
+			validation.Required,
+			is.LowerCase,
+			validation.NewStringRule(govalidator.IsDNSName, fmt.Sprintf("invalid domain name: '%s'", c.ClusterRootDomain))),
```
Comment (deifyed) [inline pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go:70]: I normally use `is.DNSName`. 

If you need to control the error message, you can use `.Error()` as in, `is.DNSName.Error("")`.

This way we don't have to directly depend on govalidator

**Review by deifyed — APPROVED** [review]
Nice!

**Review by ivaruf — APPROVED** [review]

File: pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go (hunk 1 above)
Comment (yngvark) [reply pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go]: Fixed!

**Pull request merged**